### DISCUSSION TITLE
Get player position

### DISCUSSION THREAD

**Discussion by timhc22** [pmndrs/ecctrl#94] [Q&A] [answered]
Thanks for adding the discussions section!

Is there a way to access the player position? (I'm wanting to make interactive elements for which I need this).

I've tried adding a ref to the gltf within the Ecctrl component (and to the ecctrl component itself), but ref.current.position seems to stay fixed at 0,0,0

I cant see anything in the hooks for getting this.

The only way I can think of at the moment, is to get the camera position, but that changes if the user 'zooms out' using the scrollwheel!

**Comment by ErdongChen-Andrew** [comment]
Add the ref to Ecctrl, and the position is: `ecctrlRef.current.translation()`

**Reply by timhc22** [reply]
perfect lovely, any idea if it is possible to set the position if they go past a certain point? I'm trying to figure out how to stop the player falling through the floor occasionally on load (which I assume is to do with the player loading before the model or something).

I'm also trying setting the autoBalance to false as suggested here: https://github.com/pmndrs/ecctrl/issues/48

**Reply by ErdongChen-Andrew** [reply, marked as answer]
```js
useFrame(() => {
    if (ecctrlRef.current && ecctrlRef.current.translation().y < -10) {
      ecctrlRef.current.setTranslation({ x: 0, y: 3, z: 0 });
      ecctrlRef.current.setLinvel({ x: 0, y: 0, z: 0 });
      ecctrlRef.current.setAngvel({ x: 0, y: 0, z: 0 });
    }
  });
```
https://github.com/pmndrs/ecctrl/issues/48 This is for better performance, not to prevent falling. 
Here are a few other things you could try:
1. Place the map component below the Ecctrl component.
2. Make the floor thicker.
3. Pause the physics (or gravity) until the map is fully loaded.
4. Position the character slightly higher.

**Reply by timhc22** [reply]
Great perfect thanks, I actually found a bit of a workaround by just force reloading the canvas element if the player y position was less than 0, which seems to work not too bad, but useful to see how to set the player position now too.

Do you mind if I add some of these things to the Readme, would be good to be able to help out a bit somehow!

One more question for now related to this, if I try and store the player position in the app state (or Zustand store):
```
  const [playerPosition, setPlayerPosition] = React.useState();

  useFrame(() => {
    if (playerControlsRef.current) {

      const _playerPosition = playerControlsRef.current.translation();
      setPlayerPosition(_playerPosition);
```

The player seems to continue sliding as if on Ice. I'm not sure why this is happening because I'm not doing any kind of actual player position update (as I would be in `ecctrlRef.current.setTranslation`)

Ideally I want to stream the player position to a server, and then pull down someone else's player position so that I can make multiplayer experiences, but that is a whole other set of research and questions for another discussion I'm sure!
